### PR TITLE
Implement requesting permission for using the GC Wii U Adapter.

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This gets it to the point that users will be able to use the GC Adapter as soon as they set the SIDevice to the Wii U Gamecube controller adapter.

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3553)

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java [hunk 1]
```diff
@@ -23,6 +28,26 @@
 	static UsbEndpoint usb_in;
 	static UsbEndpoint usb_out;
 
+	private static void RequestPermission()
+	{
+		HashMap<String, UsbDevice> devices = manager.getDeviceList();
+		Iterator it = devices.entrySet().iterator();
+		while (it.hasNext())
```
Comment (BhaaLseN) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java]: Any particular reason not to use a for-each loop here?

Comment (Sonicadvance1) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java]: Because I'm terrible at java and java related accessories.

Comment (BhaaLseN) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java]: `for (Entry<String, UsbDevice> pair : devices.entrySet())`

**Comment by Sonicadvance1** [comment]
Reviewed 1 of 2 files at r3.
Review status: 1 of 4 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553)*

**Comment by lioncash** [comment]
Review status: 0 of 4 files reviewed at latest revision, 4 unresolved discussions.

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/USBPermService.java, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8kKqR9gM2ibl7BNWAy:-K8kKqR9gM2ibl7BNWAz:-2073059041) ([raw file](https://github.com/dolphin-emu/dolphin/blob/394ef0ed5ed822988308c1beebae3eb624c60b53/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/USBPermService.java#L6)):*
final class

brace on new line

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java, line 39 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8kKnDxb-IdVVt597F6:-K8kKnDxb-IdVVt597F7:284638214) ([raw file](https://github.com/dolphin-emu/dolphin/blob/394ef0ed5ed822988308c1beebae3eb624c60b53/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java#L39)):*
Should be braced

---

*[Source/Core/InputCommon/GCAdapter_Android.cpp, line 38 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8kKzhFNm8SmLhaAu7V:-K8kKzhG8LN_hQ5avlZv:835612923) ([raw file](https://github.com/dolphin-emu/dolphin/blob/394ef0ed5ed822988308c1beebae3eb624c60b53/Source/Core/InputCommon/GCAdapter_Android.cpp#L38)):*
This should have an {0} initializer on it. std::atomic's default constructor doesn't provide a guarantee of automatically setting to zero.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553)*

**Comment by Sonicadvance1** [comment]
*[Source/Core/InputCommon/GCAdapter_Android.cpp, line 38 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8kKzhFNm8SmLhaAu7V:-K8kLVs4fxbKfzP_jH4q:-1892126169) ([raw file](https://github.com/dolphin-emu/dolphin/blob/394ef0ed5ed822988308c1beebae3eb624c60b53/Source/Core/InputCommon/GCAdapter_Android.cpp#L38)):*
We don't rely on its initialization value.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553)*

**Comment by lioncash** [comment]
Review status: 0 of 4 files reviewed at latest revision, 4 unresolved discussions.

---

*[Source/Core/InputCommon/GCAdapter_Android.cpp, line 38 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8kKzhFNm8SmLhaAu7V:-K8kLon8KKte8QuWuhpk:1837953364) ([raw file](https://github.com/dolphin-emu/dolphin/blob/394ef0ed5ed822988308c1beebae3eb624c60b53/Source/Core/InputCommon/GCAdapter_Android.cpp#L38)):*
Right, but your constructs should always be in a valid initial state regardless; if behavior ever changes in the future, then you don't have to care about potential uninitialized reads.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553)*

**Comment by BhaaLseN** [comment]
Reviewed 1 of 3 files at r1, 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/USBPermService.java, line 11 \[r4\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8nOOe6tchSBEfL1frv:-K8nOOe7hPpOsUG_iOsr:-1874681251) ([raw file](https://github.com/dolphin-emu/dolphin/blob/391b0aee266bf2ccd83fbc8417867fa68e02e4b7/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/services/USBPermService.java#L11)):*
Intentionally left blank? Maybe leave a note about that (and why)

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java, line 35 \[r4\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8nOX-zCZV_MYOx2UWo:-K8nOX-zCZV_MYOx2UWp:434490486) ([raw file](https://github.com/dolphin-emu/dolphin/blob/391b0aee266bf2ccd83fbc8417867fa68e02e4b7/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java#L35)):*
I don't think you still need this iterator anymore (also the `import` on top)

---

*[Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java, line 67 \[r4\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553#-K8nOe60wZyJa3GtNN06:-K8nOe60wZyJa3GtNN07:2082814963) ([raw file](https://github.com/dolphin-emu/dolphin/blob/391b0aee266bf2ccd83fbc8417867fa68e02e4b7/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/Java_GCAdapter.java#L67)):*
I'd personally prefer this to be braced too, since it isn't a single statement that follows. +/- if I'm the only one tho.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553)*

**Comment by Sonicadvance1** [comment]
Reviewed 1 of 3 files at r1, 1 of 3 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3553)*

**Pull request merged**